### ISSUE TITLE
All army units should be in the production queue only if they meet their production condition

### ISSUE THREAD

**Issue by RaphaelRoyerRivard** [RaphaelRoyerRivard/MicroMachine#719]
When that condition is no longer met, they should be removed from the queue.

**Comment by Unusual229** [comment]
We need them in the queue if we want to rebuild whatever we lost, so we can rebuild them.

**Comment by RaphaelRoyerRivard** [comment]
Units in the queue have no link with the units we lost. What I meant was: let's say we want to produce Banshees so we add them in the queue every time the queue contains none. Then, we detect that the opponent is producing Tempests, so we want to stop the Banshee production, but we still have a Banshee in the queue, so one more will be produced even though the production condition of the Banshees is not met.

**Comment by Unusual229** [comment]
Honestly, I want to get rid of the queue all together. Wouldn't it be better if we just created whatever we wanted to build directly when we have the resources?
Because we limit the queue to have a max of 1 of each type, we cannot start 2 SCVs or 2 Marines, or whatever, on the same frame, there is a maximum of 1 per frame. It also makes things more complicated than they really should be.
Unless you find something positive the queue brings us, I'd rather get rid of the queue.

**Comment by RaphaelRoyerRivard** [comment]
I agree that it is certainly not the best way of choosing what to produce, but I believe we should have an army composition analyser first and then when we really know that the units we want to produce are the right answer, then we can start to think about how to produce them in a more optimized way.
